processors/metadata.go

```diff
@@ -0,0 +1,30 @@
+package processors
+
+import "github.com/algorand/indexer/plugins"
+
+// ProcessorMetadata returns fields relevant to identification and description of plugins.
+type ProcessorMetadata struct {
+	ImplementationName        string
+	ImplementationDescription string
+	ImplementationDeprecated  bool
```

Comment:
This interface doesn't really make sense to me. Why are there getters if the fields are public? I can understand the getters if we want to make them immutable, but not otherwise. I'd consider two changes:
1) Remove `Implementation` prefix, and make the fields private.
2) You'd need a `MakeProcessorMetadata` function now to set the private fields.